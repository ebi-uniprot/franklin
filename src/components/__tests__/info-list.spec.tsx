import { render } from '@testing-library/react';

import InfoList from '../info-list';

describe('InfoList component', () => {
  test('should render', () => {
    const infoData = [
      {
        title: 'Item 1',
        content: <div>Some content</div>,
      },
      {
        title: 'Another item',
        content: <div>Some more content</div>,
      },
    ];
    const { asFragment } = render(<InfoList infoData={infoData} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render compact', () => {
    const infoData = [
      {
        title: 'Item 1',
        content: <div>Some content</div>,
      },
      {
        title: 'Another item',
        content: <div>Some more content</div>,
      },
    ];
    const { asFragment } = render(<InfoList infoData={infoData} isCompact />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render without titles', () => {
    const infoData = [
      {
        title: 'Item 1',
        content: <div>Some content</div>,
      },
      {
        title: 'Another item',
        content: <div>Some more content</div>,
      },
    ];
    const { asFragment } = render(<InfoList infoData={infoData} noTitles />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render without in 2 columns', () => {
    const infoData = [
      {
        title: 'Item 1',
        content: <div>Some content</div>,
      },
      {
        title: 'Another item',
        content: <div>Some more content</div>,
      },
    ];
    const { asFragment } = render(<InfoList infoData={infoData} columns />);
    expect(asFragment()).toMatchSnapshot();
  });
});
