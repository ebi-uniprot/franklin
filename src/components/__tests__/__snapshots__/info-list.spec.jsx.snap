// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoList component should render 1`] = `
<div
  className="info-list"
>
  <div
    className="info-list__item"
  >
    <div
      className="info-list__title"
    >
      <h5
        className="bold"
      >
        Item 1
      </h5>
    </div>
    <div
      className="info-list__content"
    >
      <div>
        Some content
      </div>
    </div>
  </div>
  <div
    className="info-list__item"
  >
    <div
      className="info-list__title"
    >
      <h5
        className="bold"
      >
        Another item
      </h5>
    </div>
    <div
      className="info-list__content"
    >
      <div>
        Some more content
      </div>
    </div>
  </div>
</div>
`;
